### PR TITLE
Week15

### PR DESCRIPTION
**Pull request by AmandaHockmuth**
Fixed security with helmet, csrf, xss-clean, and express rate limiter. Implemented CRUD functionality.

### CODE REVIEW

File: models/User.js [hunk 1]
```diff
@@ -35,19 +34,18 @@ UserSchema.pre("save", async function () {
...
 
 module.exports = mongoose.model("User", UserSchema);
+
+
+// Add team attribute to user registration
+//      - "Warning, your team cannot be changed later."
+//           if team exists add user to team
+//           if team does not exist create new team
+//           dialogue box warning "No team by that name found
+//               - create a new team?
```
Comment (jsuoz121) [inline models/User.js:50]: Are these here for your use or questions for me 😅

File: views/item.ejs [hunk 2]
```diff
@@ -0,0 +1,35 @@
+<%- include("partials/head.ejs") %>
+<%- include("partials/header.ejs") %>
+
+<form
+  method="POST"
+  action="<%- (item) ? 'http://localhost:3000/api/v1/items/update/' + item.id : 'http://localhost:3000/api/v1/items' %>">
+  <input type="hidden" name="_csrf" value="<%= _csrf %>" />
```
Comment (jsuoz121) [inline views/item.ejs:7]: Nice!

**Review by jsuoz121 — APPROVED** [review]
Really nice job! This assignment is one of the harder ones

File: models/User.js (hunk 1 above)
Comment (AmandaHockmuth) [reply models/User.js:50]: Those are notes for myself!

**Pull request merged**